modules/catalog/src/main/java/org/apache/ignite/internal/catalog/commands/altercolumn/AlterColumnNotNull.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.catalog.commands.altercolumn;
+
+import static org.apache.ignite.lang.ErrorGroups.Sql.UNSUPPORTED_DDL_OPERATION_ERR;
+
+import org.apache.ignite.internal.catalog.descriptors.TableColumnDescriptor;
+import org.apache.ignite.lang.IgniteStringFormatter;
+import org.apache.ignite.sql.SqlException;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Changes {@code nullable} flag of the column descriptor according to the {@code ALTER COLUMN (SET | DROP) NOT NULL} action.
+ */
+public class AlterColumnNotNull implements AlterColumnAction {
+    private final boolean notNull;
+
+    public AlterColumnNotNull(boolean notNull) {
+        this.notNull = notNull;
+    }
+
+    @Override
+    public @Nullable TableColumnDescriptor apply(TableColumnDescriptor origin, boolean isPkColumn) {
+        if (notNull == !origin.nullable()) {
+            return null;
```

Comment:
Null value semantic is unclear for me. E.g. I'd expect the origin descriptor should be dropped when null was returned.

Why don't just return origin if there is nothing to do? and drop @Nullable annotation at all.

Reply:
Fixed, thanks.